File: phone-bridge-ui/src/app/layout/campaign/model/CampaignPopup.ts
import { CampaignPopupIncoming } from './CampaignPopupIncoming';
import { Disposition } from './Disposition';
import { ShowField } from './ShowField';

export class CampaignPopup {
	moduleName: string;

	/* Directory URL to get info about an IC number / OG No */
	directoryInfoURL: string;

	/* No of seconds before the Popup is wrapped Up automatically */
	wrapUpTimeInSecs: number;

	campaignFeedSource: string;

	dispositions: Disposition[];
	showFields: ShowField[];

	campaignPopupIncoming: CampaignPopupIncoming;
}

File: phonebridge-portal/src/main/java/com/phonebridge/portalservice/rest/v1/row/campaign/CampaignPopup.java
package com.phonebridge.portalservice.rest.v1.row.campaign;

import java.util.List;

import lombok.Data;

@Data
public class CampaignPopup {
	
	private String moduleName;
	
	/* Directory URL to get info about an IC number / OG No */ 
	private String directoryInfoURL;
	
	/* No of seconds before the Popup is wrapped Up automatically */
	private int wrapUpTimeInSecs;
	
	private String campaignFeedSource;
	
	private List<Disposition> dispositions;
	private List<ShowField> showFields;
	
	private CampaignPopupIncoming campaignPopupIncoming;
	
	private CampaignPopupOutgoing campaignPopupOutgoing;
}
